realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -221,63 +222,26 @@ public static SyncUser login(final SyncCredentials credentials, final String aut
      * @param credentials credentials to use.
      * @param authenticationUrl server that the user is authenticated against.
      * @param callback callback when login has completed or failed. The callback will always happen on the same thread
-     *                 as this this method is called on.
+     * as this this method is called on.
+     * @return representation of the async task that can be used to cancel it if needed.
      * @throws IllegalArgumentException if not on a Looper thread.
      */
     public static RealmAsyncTask loginAsync(final SyncCredentials credentials, final String authenticationUrl, final Callback callback) {
-        if (Looper.myLooper() == null) {
-            throw new IllegalStateException("Asynchronous login is only possible from looper threads.");
-        }
-        final Handler handler = new Handler(Looper.myLooper());
-        ThreadPoolExecutor networkPoolExecutor = SyncManager.NETWORK_POOL_EXECUTOR;
-        Future<?> authenticateRequest = networkPoolExecutor.submit(new Runnable() {
+        checkLooperThread("Asynchronous login is only possible from looper threads.");
```

Comment:
Actually i met the similar problem right now with `getInstanceAsync()`, so i need a handler even before the Realm created. My currently plan is making an interface `Notifier` with `post()` method, which will only handle sending event.

Similar for the `Capabilities`, it is actually doesn't require a `SharedRealm` instance, maybe pass a `null` for now? I will change it later.